Userland/Libraries/LibC/unistd.cpp

```diff
@@ -398,11 +401,71 @@ ssize_t pwrite(int fd, const void* buf, size_t count, off_t offset)
     return nwritten;
 }
 
+// Note: Be sure to send to directory_name parameter a directory name ended with trailing slash.
+static int ttyname_r_for_directory(const char* directory_name, dev_t device_mode, ino_t inode_number, char* buffer, size_t size)
+{
+    DIR* dirstream = opendir(directory_name);
+    if (!dirstream) {
+        return -1;
+    }
+    struct stat st;
+    struct dirent* entry;
+    char* name_path;
+
+    while ((entry = readdir(dirstream)) != nullptr) {
+        if (((ino_t)entry->d_ino == inode_number)
+            && strcmp(entry->d_name, "stdin")
+            && strcmp(entry->d_name, "stdout")
+            && strcmp(entry->d_name, "stderr")) {
+
+            size_t name_length = strlen(directory_name) + strlen(entry->d_name) + 1;
+
+            if (name_length > size) {
+                if (closedir(dirstream) < 0)
+                    return -1;
```

Comment:
You can use a `ScopeGuard` here to avoid having to manually close the directory.

Reply:
So after you explained we link LibC against libstdc++ and therefore we can use C++ code, this seems like a good idea, and I will do that :)